frontend/src/components/Home.js

```diff
@@ -0,0 +1,213 @@
+        responseRates.forEach(response_rate => {
+          tractData[response_rate.tract_id] = response_rate.rate;
+        });
+        this.setState({
+          tractData: tractData
+        });
+
+        const fillColor = ["match", ["get", "GEOID"]];
+
+        // converting the response rate into a color
+        const LIGHTEST = [233, 244, 222];
+        const DARKEST = [64, 89, 34];
+        const geoIds = Object.keys(tractData);
+        geoIds.map(geoId => {
+          const rate = tractData[geoId];
+          const red = (1 - rate) * (LIGHTEST[0] - DARKEST[0]) + DARKEST[0];
+          const green = (1 - rate) * (LIGHTEST[1] - DARKEST[1]) + DARKEST[1];
+          const blue = (1 - rate) * (LIGHTEST[2] - DARKEST[2]) + DARKEST[2];
+          const color = "rgba(" + red + ", " + green + ", " + blue + ", 0.8)";
+          return fillColor.push(geoId, color);
+        });
+
+        fillColor.push("rgba(0,0,0,0)");
+
+        stateLayers.map(stateLayer => {
+          return this.map.addLayer(
+            {
+              id: stateLayer.sourceURL,
+              type: "fill",
+              source: {
+                type: "vector",
+                url: stateLayer.sourceURL
+              },
+              "source-layer": stateLayer.sourceLayer,
+              paint: {
+                "fill-color": fillColor
+              }
+            },
+            "state-label"
+          );
+        });
+      });
+    });
+
+    this.map.on("move", () => {
+      const { lng, lat } = this.map.getCenter();
+
+      this.setState({
+        lng: lng.toFixed(4),
+        lat: lat.toFixed(4),
+        zoom: this.map.getZoom().toFixed(2)
+      });
+    });
+
+    this.map.on("mousemove", e => {
+      stateLayers.forEach(element => {
+        const tracts = this.map.queryRenderedFeatures(e.point, {
+          // TODO: get all layers using a .map on stateLayers instead of hardcoding IL
+          layers: ["mapbox://meghabyte.ac7v02uw"]
+        });
+
+        if (tracts.length > 0) {
+          this.setState({
+            tractSelected: true,
+            currentTract: {
+              name: tracts[0].properties.NAMELSAD,
+              id: tracts[0].properties.GEOID
+            }
+          });
+        } else {
+          this.setState({
+            tractSelected: false,
+            currentTract: null
+          });
+        }
+      });
+    });
+  }
+
+  render() {
+    const { lng, lat, zoom, isOpen } = this.state;
+
+    return (
+      <div>
+        <div className="map">
+          <div className="inline-block absolute top right mt12 mr12 bg-darken75 color-white z1 py6 px12 round-full txt-s txt-bold">
+            <div>{`Longitude: ${lng} Latitude: ${lat} Zoom: ${zoom}`}</div>
+          </div>
+          <div
+            ref={el => (this.mapContainer = el)}
+            className={
+              isOpen
+                ? "absolute top right bottom col-9 col-s-9"
+                : "absolute top right bottom col-11 col-s-11"
```

Comment:
not sure if this is more readable, but if these classes are shared across you could also add strings instead of having the entire thing in a ternary

Reply:
this will be changed with Philip's changes